.github/workflows/cd.yaml

```
@@ -0,0 +1,51 @@
+name: Release
+on:
+  workflow_run:
+    workflows: [ ci ]
+    types:
+      - completed
+
+jobs:
+  maybe-release:
+    name: release
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && github.ref == 'refs/heads/master' && github.event.workflow_run.conclusion == 'success'
```

```suggestion
    if: github.event.workflow_run.conclusion == 'success'
```
`github.event_name == 'push'` - this won't work as event is workflow_run.
`github.ref == 'refs/heads/master'` - I left other comment